robot/vision.py

```diff
@@ -209,13 +210,29 @@ class RoboConUSBCamera(Camera):
     def __init__(self,
                  start_res=(1296, 736),
                  focal_lengths=None):
-        self._cv_capture = cv2.VideoCapture(0)
+        self._source = self.find_usb_cam()
+        if self._source == None:
+            raise Exception("No USB camera detected")
+        self._cv_capture = cv2.VideoCapture(self._source)
         self._res = start_res
         self.focal_lengths = (LOGITECH_C270_FOCAL_LENGTHS
                               if focal_lengths is None
                               else focal_lengths)
         self._update_camera_params(self.focal_lengths)
 
+    def find_usb_cam(self):
+        indicator_string = "bm2835" # if this string is present in the output then it is a pi cam
```

Comment:
MAYBE: This might change if we ever move to using `libcamera` by default which the raspberry pi foundation is moving to. Also will definely change if we ever move to a Pi not based on the BM2835's peripheral set. A more future proof way might be to find USB cameras rather than trying to find the pi camera out of the USB cameras. The phrase USB being in the `bus info` is probally a good way to do this.